extensions/who-is-visiting-this-content/app.R

```diff
@@ -0,0 +1,151 @@
+library(shiny)
+library(bslib)
+library(shinyjs)
+library(connectapi)
+library(dplyr)
+library(glue)
+library(lubridate)
+library(tidyr)
+
+shinyOptions(
+  cache = cachem::cache_disk("./app_cache/cache/", max_age = 60 * 60 * 8)
+)
+
+source("get_usage.R")
+
+ui <- page_fillable(
+  useShinyjs(),
+  theme = bs_theme(version = 5),
+
+  card(
+    card_header("Who is Visiting This Content?"),
+    layout_sidebar(
+      sidebar = sidebar(
+        title = "No Filters Yet",
+        open = FALSE,
+
+        actionButton("clear_cache", "Clear Cache", icon = icon("refresh"))
+      ),
+
+      textInput(
+        "content_guid",
+        "Content GUID"
+      ),
+
+      h4(
+        id = "guid_input_msg",
+        "Please enter a content GUID"
+      ),
+
+      textOutput("summary_message"),
+
+      tabsetPanel(
+        id = "content_visit_tables",
+        tabPanel(
+          "List of Visits",
+          tableOutput("all_visits")
+        ),
+        tabPanel(
+          "Aggregated Visits",
+          tableOutput("aggregated_visits")
+        )
+      )
+    )
+  )
+)
+
+server <- function(input, output, session) {
+  # Cache invalidation button ----
+  cache <- cachem::cache_disk("./app_cache/cache/")
+  observeEvent(input$clear_cache, {
+    print("Cache cleared!")
+    cache$reset()  # Clears all cached data
+    session$reload()  # Reload the app to ensure fresh data
+  })
+
+  observe({
+    if (nchar(input$content_guid) == 0) {
+      show("guid_input_msg")
+      hide("content_visit_tables")
+    } else {
+      hide("guid_input_msg")
+      show("content_visit_tables")
+    }
+  })
```

Comment:
I _think_ this would go here, but what do you think about adding in a parser that lets someone paste in a full URL that includes the GUID? This is a bit of an extension / scope creep so if we want to make a separate issue for it and do that in a follow on that's totally fine. But the UX of needing to select _just_ the GUID form the URL (or get to it in the settings panel) could be better. We had good success with the publisher where the field that accepts GUID also accepts the URL (and, in fact, in the publisher it's the GUID alone that is the hidden easter egg that it _just works_, we though that folks would be more confident / able to / comfortable with grabbing the URL)

Reply:
I was trying to keep this as minimal as possible, figuring it wouldn't be the final UI for selecting a piece of content.

Perhaps it would be fine to add a bit of extra logic to extract the GUID if a URL is pasted in. OTOH, perhaps even having this hide / show logic and a "paste in a content guid" message is too much code to pour into something that won't be a final UI. I can really see both arguments. I'd rather not put more effort in here till a later edition, at the very least.